### PR TITLE
Adds query support for several indexes to support pagination and other things

### PR DESCRIPTION
**Pull request by CallumEddisford**

### CODE REVIEW

File: src/Services/Mailboxes.ts [hunk 1]
```diff
@@ -8,15 +8,17 @@ import Xuid, { SupportedXuid } from "../Utils/Xuid";
...
      * Lists mailboxes
+     * @param params
      * @returns Promise<Mailbox[]>
      */
-    list() {
+    list(params: object) {
         return this.client
-            .get<JsonResponse<Collection<Mailbox>>>(`1/mailboxes`)
+            .get<JsonResponse<Collection<Mailbox>>>(`1/mailboxes`, {params})
             .then((response: AxiosResponse<JsonResponse<Collection<Mailbox>>>) => {
```
Comment (Smudge3806) [inline src/Services/Mailboxes.ts:20]: Shouldn't these be `PaginatedJsonResponse`s

File: src/Services/Products.ts [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
+     * @param params
      * @returns <Promise>Product[]>
      */
-    getByBusiness(businessId: Xuid<SupportedXuid.Business>)
-    {
+    list(params: object) {
         return this.client
-            .get<JsonResponse<Collection<Product>>>(`1/businesses/${ businessId }/products`)
+            .get<JsonResponse<Collection<Product>>>(`1/products`, {params})
             .then((response: AxiosResponse<JsonResponse<Collection<Product>>>) => {
```
Comment (Smudge3806) [inline src/Services/Products.ts:19]: Paginated?

File: src/Services/TaxRates.ts [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
+     * @param params
      * @returns <Promise>TaxRate[]>
      */
-    getByBusiness(businessId: Xuid<SupportedXuid.Business>)
-    {
+    list(params: object) {
         return this.client
-            .get<JsonResponse<Collection<TaxRate>>>(`1/businesses/${ businessId }/tax`)
+            .get<JsonResponse<Collection<TaxRate>>>(`1/tax`, {params})
             .then((response: AxiosResponse<JsonResponse<Collection<TaxRate>>>) => {
```
Comment (Smudge3806) [inline src/Services/TaxRates.ts:19]: Paginated?

**Comment by CallumEddisford** [comment]
Fixed dude

**Pull request merged**